src/nvim/fileio.c

```diff
@@ -4448,14 +4448,24 @@ bool vim_fgets(char_u *buf, int size, FILE *fp) FUNC_ATTR_NONNULL_ALL
   char tbuf[FGETS_SIZE];
 
   buf[size - 2] = NUL;
+retry:
+  errno = 0;
   eof = fgets((char *)buf, size, fp);
+  if (eof == NULL && errno == EINTR) {
+    goto retry;
+  }
   if (buf[size - 2] != NUL && buf[size - 2] != '\n') {
     buf[size - 1] = NUL;            /* Truncate the line */
 
     /* Now throw away the rest of the line: */
     do {
       tbuf[FGETS_SIZE - 2] = NUL;
+retry_ignore:
+      errno = 0;
       ignoredp = fgets((char *)tbuf, FGETS_SIZE, fp);
+      if (ignoredp == NULL && errno == EINTR) {
+        goto retry_ignore;
```

Comment:
technically we're not ignoring the retval now, so `ignoredp` should not be used.

Reply:
Yeah, I guess we can just rely on the existing loop to retry the `fgets` here.

Reply:
Wait, this won't properly handle an EOF, either...

Reply:
just to be clear, i was only speaking about the specific `ignoredp` global which is used as a dummy assignment when Vim doesn't care about some return value.